openlibrary/plugins/openlibrary/js/SearchBar.js

```diff
@@ -0,0 +1,298 @@
+        $(window).resize(debounce(() => {
+            this.toggleCollapsibleModeForSmallScreens($(window).width());
+        }, 50));
+        $(document).on('submit','.in-collapsible-mode', event => {
+            if (this.collapsed) {
+                event.preventDefault();
+                this.toggleCollapse();
+                this.$input.focus();
+            }
+        });
+    }
+
+    /**
+     * Enables/disables CollapsibleMode depending on screen size
+     * @param {Number} windowWidth
+     */
+    toggleCollapsibleModeForSmallScreens(windowWidth) {
+        if (windowWidth < 568) {
+            if (!this.inCollapsibleMode) {
+                this.enableCollapisbleMode();
+                this.collapse();
+            }
+            this.clearAutocompletionResults();
+        } else {
+            if (this.inCollapsibleMode) {
+                this.disableCollapsibleMode();
+            }
+        }
+    }
+
+    /** Collapses or expands the searchbar */
+    toggleCollapse() {
+        if (this.collapsed) {
+            this.expand();
+        } else {
+            this.collapse();
+        }
+    }
+
+    collapse() {
+        $('header#header-bar .logo-component').removeClass('hidden');
+        this.$component.removeClass('expanded');
+        this.collapsed = true;
+    }
+
+    expand() {
+        $('header#header-bar .logo-component').addClass('hidden');
+        this.$component.addClass('expanded');
+        this.collapsed = false;
+    }
+
+    enableCollapisbleMode() {
+        this.$form.addClass('in-collapsible-mode');
+        this.inCollapsibleMode = true;
+    }
+
+    disableCollapsibleMode() {
+        this.collapse();
+        this.$form.removeClass('in-collapsible-mode');
+        this.inCollapsibleMode = false;
+    }
+
+    /**
+     * Converts an already processed query into a search url
+     * @param {String} q query that's ready to get passed to the search endpoint
+     * @param {Boolean} [json] whether to hit the JSON endpoint
+     * @param {Number} [limit] how many items to get
+     */
+    composeSearchUrl(q, json, limit) {
+        let url = this.facetEndpoint;
+        if (json) {
+            url += '.json';
+        }
+        url += `?q=${q}`;
+        if (limit) {
+            url += `&limit=${limit}`;
+        }
+        url += `&mode=${SearchUtils.mode.read()}`;
+        return url;
+    }
+
+    /**
+     * Prepare an unprocessed query for book searching
+     * @param {String} q
+     * @return {String}
+     */
+    static marshalBookSearchQuery(q) {
+        if (q && q.indexOf(':') == -1 && q.indexOf('"') == -1) {
+            q = `title: "${q}"`;
+        }
+        return q;
+    }
+
+    /** Setup event listeners for autocompletion */
+    initAutocompletionLogic() {
+        // searches should be cancelled if you click anywhere in the page
+        $(document.body).on('click', this.clearAutocompletionResults.bind(this));
```

Comment:
Should be outside the responsibility of the class

Reply:
I'll explain some more. I think this is super important and if we are going to refactor this class and start pushing towards a component library, it's essential we don't do this.

Reply:
I understand where you're coming from, but I don't have a good/quick solution to address those conerns, and this isn't a regression (it was already there). Still opting for won't fix on this one, since we can tackle it at any point in the future, and it doesn't devalue the other work done here.